src/include/units/physical/si/length.h

```diff
@@ -34,8 +34,34 @@ struct centimetre : prefixed_unit<centimetre, centi, metre> {};
 struct decimetre : prefixed_unit<decimetre, deci, metre> {};
 struct hectometre : prefixed_unit<hectometre, hecto, metre> {};
 struct kilometre : prefixed_unit<kilometre, kilo, metre> {};
+struct femtometre : prefixed_unit<femtometre,femto,metre> {};
 
 struct astronomical_unit : named_scaled_unit<astronomical_unit, "au", no_prefix, ratio<149'597'870'700>, metre> {};
+#if 0
+struct inch : named_scaled_unit<inch,"in(imp)",no_prefix,ratio<127,50,-2>, metre> {};
+struct foot : named_scaled_unit<foot,"ft(imp)",no_prefix,ratio<381,125,-1>, metre> {};
+struct yard : named_scaled_unit<yard,"yd(imp)",no_prefix,ratio<1143,125,-1>, metre> {};
+#endif
+struct foot_us : named_scaled_unit<foot_us,"ft(us)",no_prefix,ratio<1524003,500000,-1>, metre> {};
+struct fathom : named_scaled_unit<fathom,"fathom(imp)",no_prefix,ratio<1143,625>, metre> {};
+struct fathom_us : named_scaled_unit<fathom_us,"fathom(us)",no_prefix,ratio<457201,250000>,metre> {};
+struct chain : named_scaled_unit<chain,"ch(us)",no_prefix,ratio<502921,250000,1>, metre> {};
+
+#if 0
+struct thou : named_scaled_unit<thou,"thou",no_prefix,ratio<127,50,-5>,metre> {};
+#endif
+struct mil : named_scaled_unit<mil,"mil",no_prefix,ratio<127,50,-5>,metre> {};
+
+struct light_year : named_scaled_unit<light_year,"ly(iau)",no_prefix,ratio<946073,100000,15>,metre> {};
+struct mile : named_scaled_unit<mile,"mi(imp)",no_prefix,ratio<25146,15625,3>,metre> {};
+struct nautical_mile : named_scaled_unit<nautical_mile,"mi(naut)",no_prefix, ratio<463,250,3>,metre> {};
+struct parsec : named_scaled_unit<parsec,"pc(iau)",no_prefix,ratio<1542839,500000,16>,metre> {};
+struct pica_comp : named_scaled_unit<pica_comp,"pica(comp)",no_prefix,ratio<4233333,1000000,-3>,metre> {};
+struct pica_prn : named_scaled_unit<pica_prn,"pica(prn)",no_prefix,ratio<2108759,500000,-3>,metre> {};
+struct point_comp : named_scaled_unit<point_comp,"point(comp)",no_prefix,ratio<1763889,500000,-4>,metre> {};
+struct point_prn  : named_scaled_unit<point_prn,"point(prn)", no_prefix, ratio<1757299,500000,-4>,metre> {};
+struct rod : named_scaled_unit<rod,"rd(us)",no_prefix, ratio<502921,100000>,metre> {};
+struct angstrom : named_scaled_unit<angstrom,"angstrom(nru)",no_prefix,ratio<1,1,-10>,metre> {};
```

Comment:
I think you have imperial conversion factor unit for foot_us. Or at least for one cant rememeber which.

Useful link
https://www.nist.gov/pml/special-publication-811/nist-guide-si-appendix-b-conversion-factors/nist-guide-si-appendix-b8

Reply:
If it is true, then we have to fix it. We should also add imperial units there, but for now, I scoped mostly on the frameworks rather than coverage.

Reply:
[Apparently US foot  is not always 1/3rd of a yard](https://www.nist.gov/pml/special-publication-811/nist-guide-si-appendix-b-conversion-factors#B6).

I have international ft ratio<381,125,-1>  --> metre
https://github.com/kwikius/quan-trunk/blob/master/quan/components/of_length.hpp#L96
us ft ( us survey foot?)  ratio<1524003,500000,-1>  --> metre
https://github.com/kwikius/quan-trunk/blob/master/quan/components/of_length.hpp#L104

There is a good question here as to whether these systems really require their own namespaces or are actually simply conversion factors that can be put in a single legacy conversion factors namespace.

Reply:
Really interesting, thanks! Anyway, I would vote for removing US (and other i.e. imperial units) from SI subdirectory to keep it pure and simple (also for standardization purposes). Please use `us` subdirectory for Customary Units and create additional directories (i.e. imperial) for others.

Reply:
Really interesting, thanks! Anyway, I would vote for removing US (and other i.e. imperial units) from SI subdirectory to keep it pure and simple (also for standardization purposes). Please use `us` subdirectory for Customary Units and create additional directories (i.e. `imperial`) for others.

Reply:
> Really interesting, thanks! Anyway, I would vote for removing US (and other i.e. imperial units) from SI subdirectory to keep it pure and simple (also for standardization purposes). Please use `us` subdirectory for Customary Units and create additional directories (i.e. `imperial`) for others.

Unfortunately the problem of who owns which unit is not simple. Wars have been started over less.... ;)
https://www.dailymail.co.uk/news/article-3967772/Staunch-Brexiteer-82-spent-past-16-years-travelling-country-removing-hated-EU-road-signs-showing-distances-metres.html

Figuring out what belongs to who is a big job. How about I just put them in  a 'lost_and_found" namespace for now and the advocates for these systems can fight over them?

Reply:
LOL ;-) Please do not just put everything to `lost_and_found`. We can try to divide them at least to `us` and `imperial` (in case of issues [let's do like others do](https://www.boost.org/doc/libs/1_72_0/doc/html/boost_units/Reference.html#boost_units.Reference.base_units_by_category)). If something will be left and claimed necessary we can pu ti to `others` for now.